File: backend/data/generated-driver-examples/binary-search/driver.py
import sys
import json
from typing import List, Optional, Any


def main():
    lines = [line.strip() for line in sys.stdin if line.strip()]
    if not lines:
        return
    
    solution = Solution()
    idx = 0
    while idx < len(lines):
        nums = json.loads(lines[idx + 0])
        target = int(lines[idx + 1])
        result = solution.search(nums, target)
        print(result)
        idx += 2

if __name__ == "__main__":
    main()

File: backend/data/generated-driver-examples/binary-search/driver.js
const fs = require('fs');

function main() {
  const input = fs.readFileSync(0, 'utf-8');
  const lines = input.split('\n').map(l => l.trim()).filter(l => l.length > 0);
  if (lines.length === 0) return;

  let idx = 0;
  while (idx < lines.length) {
    var nums = JSON.parse(lines[idx + 0]);
    var target = parseInt(lines[idx + 1], 10);
    var result = search(nums, target);
    console.log(result);
    idx += 2;
  }
}

main();